cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -153,61 +198,98 @@ private void updateProfileMetadata(EntityId entityId, MetadataMessage message) t
 
   private void collectProfileMetadata(EntityId entityId, MetadataMessage message,
                                       List<MetadataMutation> updates) throws IOException {
+    NamespaceId namespaceId = null;
```

Comment:
All these variables seems like they are local to each of the case, right? In that case, it is better to move it closer to where it is being used (i.e. inside the case block).